File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int maxDepth = 0;
    public void helper(TreeNode root, int level){
        if(root == null) return;
        if(root.left == null && root.right == null){
            maxDepth = Math.max(level, maxDepth);
        }
        level++;
        helper(root.left,level);
        helper(root.right,level);
    }
    public int maxDepth(TreeNode root) {
        helper(root, 1);
        return maxDepth;
    }
}

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        dq = deque()
        dq.append((root, 1))
        depth = 1
        while dq:
            node, level = dq.popleft()
            depth = max(depth, level)

            if node.left:
                dq.append((node.left, level+1))

            
            if node.right:
                dq.append((node.right, level+1))

        return depth
        